src/server.coffee

```diff
@@ -591,3 +597,25 @@ else
           type: 'restart-all'
       , 2000
     done()
+
+  # function to return process uptimes
+  exports.getUptime = (callback) ->
+
+    if cluster.isMaster
+      # send reponse back to API request
+      uptime =
+        master: process.uptime()
+      callback null, uptime
+    else
+      # send request to master
+      process.send
+        type: 'get-uptime'
+
+      # listen for response from master
+      process.once 'message', (uptime) ->
+        if uptime.type is 'get-uptime'
```

Comment:
If the event listener goes into the 'else' then the request wouldnt have come from the heartbeat API request. Would this then ever return null to the heart API call?